src/CiDetector.php

```diff
@@ -30,7 +30,7 @@ class CiDetector
     /** @var Env */
     private $environment;
 
-    public function __construct()
+    final public function __construct()
```

Comment:
Why not the entire class?

Reply:
I don't think its necessary to limit usage of the class here. I'm ok with having the class open for extension if someone needs to (and the interface is not enough for whatever reason). As far as they are not able to break the constructor.